### ISSUE TITLE
Adding shift leads to departments

### ISSUE THREAD

**Issue by itsrachelfish** [playasoft/volunteers#20]
In the current Apogaea Volunteer Database, it's possible to add shift leads to a department. These leads then have access to change the options of the department, create new shifts, and kick users from slots. This would require some pretty major changes in laravel-voldb's system for handling roles.

**Comment by IzKen** [comment]
I don't think the 2016 VolDB has this type of support. I only see "Staff status" option on users, which makes them an admin.

So I think it only has user & admin level support. Yes, having more granular roles is needed, but lower priority  for 2017 I think.  For this year we can make those who need to add\edit shift admins I think. This will mean we might have around 30 admins that can screw stuff up, so I would say admin rights would only be given out on an as needed basis and only after they had some "training."

This could be moved to 2018 Milestone, but at least a lower priority for 2017.

**Comment by itsrachelfish** [comment]
A `department-lead` permission was added to the system as of 596f5fb4db1af7c1ec9d7ccada948441f9975054
